ocs_ci/deployment/node_utils.py

```diff
@@ -0,0 +1,488 @@
+"""
+This module contains classes which provides support
+for addition, maintenance of nodes
+"""
+import os
+import logging
+
+import boto3
+import pytest
+
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import ocp, constants, exceptions
+from .node import AWSNode, VMWareNode
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.utils import (
+    get_infra_id, create_rhelpod, get_ocp_version,
+    TimeoutSampler
+)
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility import templating
+
+logger = logging.getLogger(__name__)
+
+node_cls_map = {'aws': AWSNode, 'vmware': VMWareNode}
+
+
+class NodeUtils(object):
+    def __init__(self):
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        self.platform = config.ENV_DATA['platform']
+        self.cluster_path = config.ENV_DATA['cluster_path']
+        self.cluster_id = get_infra_id(self.cluster_path)
+        if 'aws' in self.platform:
+            self.region = config.ENV_DATA['region']
+
+    def prepare_nodes(self, node_config, node_t=None, num_nodes=1):
+        """
+        Prepare multiple node instances
+
+        Args:
+            node_config (dict): node config
+            node_t (str): type of node eg: RHEL/RHCOS etc
+            num_nodes (int): Number of nodes to be created
+
+        Returns:
+            list: of node objects
+
+        """
+        node = node_cls_map[config.ENV_DATA['platform']]
+        nlist = []
+        num_workers = int(os.environ.get('num_workers', 3))
+        for i in range(num_nodes):
+            if not node_config.get('zone_worker_id'):
+                node_config['zone_worker_id'] = (i % num_workers)
+            nlist.append(node(node_config, node_t))
+            nlist[i].prepare_node()
+        return nlist
+
+    def attach_nodes_to_cluster(self, node_list):
+        """
+        Prepare aws nodes with given node_config
+
+        Args:
+            node_list(list): of Node objects
```

Comment:
Empty line at the end of the docstring here, same for some of the other methods in this class.